SimTracker/SiStripDigitizer/plugins/SiLinearChargeDivider.cc

```diff
@@ -19,8 +19,32 @@ SiLinearChargeDivider::SiLinearChargeDivider(const edm::ParameterSet& conf) :
   //for the cosimc generation (Considering only the tracker the value is 11 ns)
   cosmicShift(conf.getUntrackedParameter<double>("CosmicDelayShift")),
   theParticleDataTable(nullptr),
+  resolution(peakMode?(conf.getParameter<double>("resolutionPeak")):(conf.getParameter<double>("resolutionDeco"))),
   // Geant4 engine used to fluctuate the charge from segment to segment
-  fluctuate(new SiG4UniversalFluctuation()) {
+  fluctuate(new SiG4UniversalFluctuation())
+
+  
+{
+  std::ifstream APVShapeReadFile(peakMode?(conf.getParameter<edm::FileInPath>("APVShapePeakFile").fullPath().c_str()):(conf.getParameter<edm::FileInPath>("APVShapeDecoFile").fullPath().c_str()));
+  if (!APVShapeReadFile.good()){ 
+    if(peakMode){
+      edm::LogError("FileError") << "Problem opening APV Shape file: "<<conf.getParameter<edm::FileInPath>("APVShapePeakFile").fullPath();
+      return;
+    }
+    else {
+      edm::LogError("FileError") << "Problem opening APV Shape file: "<<conf.getParameter<edm::FileInPath>("APVShapeDecoFile").fullPath();
```

Comment:
Would it make sense to throw an exception instead? (or can one do a meaningful simulation without this information?)